### DISCUSSION TITLE
Can't get set_text_voice_synced method to work

### DISCUSSION THREAD

**Discussion by uptothedog** [dialogic-godot/dialogic#2194] [Q&A] [answered]
Hi, I'm trying to sync up my text reveal speed to my voice audio files as this seems to make a lot of sense in my use case. In the [documentation](https://docs.dialogic.pro/classes/subsystem_text.html#method-set_text_voice_synced) I've found the following built-in method, which should exist for the Text subsystem:

> [func ](https://docs.dialogic.pro/classes/subsystem_text.html#method-set_text_voice_synced)[set_text_voice_synced](https://docs.dialogic.pro/classes/subsystem_text.html#method-set_text_voice_synced) ( enabled: [bool](https://docs.godotengine.org/en/latest/classes/class_bool.html#class-bool) = true ) ⇒ void
> 
> This method will sync the text speed to the voice audio clip length, if a voice is playing. For instance, if the voice is playing for four seconds, the text will finish revealing after this time. This feature ignores Auto-Pauses on letters. Pauses via BBCode will desync the reveal.

Now it's fair to say I'm a total beginner so I might just be messing this up, but I'm failing to find any trace of this built-in method even existing in the Text subsystem by trying to call it, nor do I find a mention of it within the `subsystem_text.gd` script.

Could someone give me some pointers on how to use it?

**Comment by CakeVR** [comment]
Hello, you would call this as `Dialogic.Text.set_text_voice_synced(true)`. This is the same procedure for all the methods on that page.

**Reply by uptothedog** [reply]
Hey CakeVR, I've tried that but got the error:

`Invalid call. Nonexistent function 'set_text_voice_synced' in base 'Node (subsystem_text.gd)'.`

**Reply by CakeVR** [reply]
What's your Dialogic 2 version? It should have been added to Alpha 13.

**Reply by uptothedog** [reply]
Strange! I do have Alpha 13...

Maybe my files just aren't quite right. I think my project originally had Dialogic 1 and I upgraded it.

I'll have to compare -- thanks for the information!

**Reply by uptothedog** [reply]
Not sure why, but using the Alpha 13 files straight off here are giving me some errors. This happens as I'm trying to load the premade Visual Novel Style or anything custom of mine that's based on it.

` scene/resources/resource_format_text.cpp:595 - res://Scenes/VisualNovelTextbox/vn_textbox_default_panel.tres:26 - Parse Error: 
  Failed loading resource: res://Scenes/VisualNovelTextbox/vn_textbox_default_panel.tres. Make sure resources have been imported by opening the project in the editor at least once.`


 ` scene/resources/resource_format_text.cpp:284 - res://addons/dialogic/Modules/DefaultLayoutParts/Layer_VN_Textbox/vn_textbox_layer.tscn:273 - Parse Error: 
  Failed loading resource: res://addons/dialogic/Modules/DefaultLayoutParts/Layer_VN_Textbox/vn_textbox_layer.tscn. Make sure resources have been imported by opening the project in the editor at least once.`

  `scene/resources/resource_format_text.cpp:595 - res://addons/dialogic/Modules/DefaultLayoutParts/Style_VN_Default/default_vn_style.tres:26 - Parse Error: 
  Failed loading resource: res://addons/dialogic/Modules/DefaultLayoutParts/Style_VN_Default/default_vn_style.tres. Make sure resources have been imported by opening the project in the editor at least once.`

`res://addons/dialogic/Modules/StyleEditor/style_editor.gd:230 - Attempt to call function 'clone' in base 'null instance' on a null instance.`

**Reply by uptothedog** [reply]
The above seems to have possibly been caused by a corrupt style or something like that. I'm still not sure what happened, but eventually I was able to make a new style from the default one again.

After extracting the newer files however it seems my signals are no longer working. The error suggests there is no Voice subsystem, which there is... Right?
`Dialogic.Voice.connect("voiceline_finished", Callable(self, "_on_voiceline_finished"))`

`Cannot call method 'connect' on a null value. E 0:00:03:0002   DialogicGameHandler.gd:346 @ get_subsystem(): Node not found: "Voice" (relative to "/root/Dialogic").`

**Reply by CakeVR** [reply]
If you installed Dialogic 1 before, you may have indeed caused some corruption.
I would recommend creating a new project, migrating your files into it, and cleanly installing Dialogic 2 into that.

**Reply by uptothedog** [reply]
Hmm, that's a shame; I did try to follow the instructions closely, and did use the migration tool. Thankfully I'm still in the beginnings of my game project, so it shouldn't take too long to do a clean install as you described.

I don't know if my experience will be helpful or not (I know those errors aren't very specific), but with the official release of Dialogic 2, there will be more users in the near future making the migration from 1 to 2.

To possibly help prepare for issues that may arise, let me know if there's any specific files you'd like to look at that have been created as a result of version 1 to 2 migration, and I can then try and provide those (I'll be keeping the malfunctioning project).

Thank you for your help.

**Reply by CakeVR** [reply]
I don't recommend changing from Dialogic 1 to 2. First, Godot 3 and 4 (requirement for Dialogic 2) are fundamentally different and so are Dialogic 1 and 2. I never used the existing Converter, we moved it out of the main repository, and there is currently no more development planned for it. It may still work to degrees. There are some success stories using the Converter.

You don't have to create a new project, but it's the cleanest way to go about this. The other way would require to cleanly check all the configuration files of Godot and clear the `.godot` folder cache.

**Reply by uptothedog** [reply]
Well, I've now migrated to a new project, but it seems this signal still isn't working. Is my way of connecting the signal incorrect? And if so, has the way Dialogic emits these signals perhaps changed somewhere along the line, as this connection did work before?

To provide further context, I have a script to handle the animations of my custom portrait. I've set this script to wait for signals from Dialogic, and my goal with these specific signals is to revert my portrait to an idle animation when the voice line is no longer playing.

Dialogic Core is set to autoload for the project.

Dialogic signal in the portrait handler script:
`Dialogic.Voice.connect("voiceline_finished", Callable(self, "_on_voiceline_finished"))`

Commenting anything out after when the script is supposed to be receiving the signal doesn't affect Godot in throwing me the error.

Error:

Stack trace: `Cannot call method 'connect' on a null value.`
Errors:
```
E 0:00:01:0990   DialogicGameHandler.gd:346 @ get_subsystem(): Node not found: "Voice" (relative to "/root/Dialogic").
  <C++ Error>    Method/function failed. Returning: nullptr
  <C++ Source>   scene/main/node.cpp:1638 @ get_node()
  <Stack Trace>  DialogicGameHandler.gd:346 @ get_subsystem()
                 DialogicGameHandler.gd:129 @ @Voice_getter()
                 PortraitHandler.gd:8 @ _ready()
```

If this path is supposed to be the project path, it looks incorrect in how Dialogic 2 is supposed to be installed (in the /addons/ folder).

![Screenshot 2024-04-19 131709](https://github.com/dialogic-godot/dialogic/assets/88539695/e1b0e79c-1292-4d3e-8662-21e3eae2b3f7)

![Screenshot 2024-04-19 131812](https://github.com/dialogic-godot/dialogic/assets/88539695/18bfb909-3911-4b37-852f-1fd89d2a968e)

**Reply by uptothedog** [reply, marked as answer]
Solved. I'd made a mistake with the loading order and the signals weren't there when the script was trying to connect to them, because they didn't get the chance to exist yet.

It seems this would have been very difficult to infer from the information I'd shared, so that'd be a lesson for me. My bad!
